R/format_data.R

```diff
@@ -2414,6 +2414,331 @@ fmt_datetime <- function(data,
   )
 }
 
+#' Format values as time durations
+#'
+#' @description
+#' Format input values to time durations whether those input values are numbers
+#' or of the `difftime` class. We can specify which time units any numeric input
+#' values have (as days, hours, minutes, or seconds) and the output can be
+#' customized with a duration style (corresponding to short, long, and
+#' colon-separated forms) and a choice of output units from weeks to seconds.
+#'
+#' @details
+#' Targeting of values is done through `columns` and additionally by `rows` (if
+#' nothing is provided for `rows` then entire columns are selected). Conditional
+#' formatting is possible by providing a conditional expression to the `rows`
+#' argument. See the *Arguments* section for more information on this.
+#'
+#' @inheritParams fmt_number
+#' @param input_units If one or more selected columns contains numeric values,
+#'   the keyword provided for `input_units` will determine how those values are
+#'   interpreted in terms of a single duration type. If numeric columns are
+#'   included in `columns` then one of the following keywords, representative of
+#'   time units, must be provided: seconds (`"secs"`), `"mins"` (minutes),
+#'   `"hours"`, `"days"`, or `"weeks"`.
+#' @param output_units The output values can be handled by one of two keywords
+#'   (`"auto"`, the default, or `"all"`) or a vector of the following time
+#'   parts: `"weeks"`, `"days"`, `"hours"`, `"mins"`, or `"secs"`.
+#' @param duration_style A choice of three formatting styles for the output
+#'   duration values. With `"narrow"` (the default style), durations will be
+#'   formatted with single letter time-part units (e.g., 1.35 days will be
+#'   styled as `"1d 8h 24m`). With `"wide"`, this example value will be expanded
+#'   to `"1 day 8 hours 24 minutes"` after formatting. The `"colon-sep"` style
+#'   of formatting will separate all values with a colon and will not provide
+#'   any time-part units, so the output of 1.35 days will be `"1:8:24"`.
+#' @param drop_zero_parts An option to drop leading and internal values for
+#'   time parts if they are zero. By default this is `TRUE` and duration values
+#'   that might otherwise be formatted as `"0w 1d 0h 4m 19s"` with
+#'   `drop_zero_parts = FALSE` are instead displayed as `"1d 4m 19s"`.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @family Format Data
+#' @section Function ID:
+#' 3-11
+#'
+#' @import rlang
+#' @export
+fmt_duration <- function(
+    data,
+    columns,
+    rows = everything(),
+    input_units = NULL,
+    output_units = "auto",
```

Comment:
The current definition of `output_units` (taking a vector of different time parts) lets the user potential specify a discontiguous list of time parts (`c("weeks", "hours")`). We can sidestep this problem by only taking:

1. `NULL` - same as "all" is now
2. 1 element - use only this unit
3. 2 elements - use this as the range of units (e.g. `c("weeks", "hours")` would mean `c("weeks", "days", "hours")`)